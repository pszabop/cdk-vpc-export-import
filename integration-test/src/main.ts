import * as Ec2 from '@aws-cdk/aws-ec2';
import * as Cdk from '@aws-cdk/core';
import { VpcPortable } from '../../src';
// @ts-ignore
import * as util from 'util'; // eslint-disable-line

const exportingStackName = 'exportStack';
const importingStackName = 'importStack';
const testVpcName = 'testVpc'

export class SrcStack extends Cdk.Stack {
  constructor(scope: Cdk.Construct, id: string, props: Cdk.StackProps = {}) {
    super(scope, id, props);

    // define resources here...
    const testVpc = new VpcPortable(this, testVpcName, {
      subnetConfiguration: [
        {
          cidrMask: 22,
          name: 'isolated',
          subnetType: Ec2.SubnetType.ISOLATED,
        },
        {
          cidrMask: 22,
          name: 'private',
          subnetType: Ec2.SubnetType.PRIVATE,
        },
        {
          cidrMask: 22,
          name: 'public',
          subnetType: Ec2.SubnetType.PUBLIC,
        },
      ],
    });

    testVpc.exportToStackOutput();
  }
}

export class DestStack extends Cdk.Stack {
  constructor(scope: Cdk.Construct, id: string, props: Cdk.StackProps = {}) {
    super(scope, id, props);

    // define resources here...
    const corporatedVpc = VpcPortable.importFromStackOutput(this, 'corporatedVpc', exportingStackName, testVpcName);
    console.log(`typeof: ${typeof corporatedVpc}`);

  }
}
// for development, use account/region from cdk cli
//
/*
const devEnv = {
  account: process.env.CDK_DEFAULT_ACCOUNT,
  region: process.env.CDK_DEFAULT_REGION,
};
*/

const app = new Cdk.App();

new SrcStack(app, exportingStackName);
new DestStack(app, importingStackName);
//new SrcStack(app, 'my-stack-dev', { env: devEnv });
// new SrcStack(app, 'my-stack-prod', { env: prodEnv });

app.synth();
