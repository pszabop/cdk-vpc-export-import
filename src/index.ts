import * as Ec2 from '@aws-cdk/aws-ec2';
import * as Cdk from '@aws-cdk/core';

export class VpcPortable extends Ec2.Vpc {


  static mangleStackName(name: string) {
    return name.replace(/-|\s/g, '');
  }

  static getOtherStackOutput(key: string, otherStackName: string, otherStackNodeId: string): string {
    const globallyUniqueKeyExport = `${VpcPortable.mangleStackName(otherStackName)}-${otherStackNodeId}-${key}`;
    const result = Cdk.Fn.importValue(globallyUniqueKeyExport);
    return result;
  }

  static getOtherStackOutputList(key: string, otherStackName: string, otherStackNodeId: string): string[] {
    const globallyUniqueKeyExport = `${VpcPortable.mangleStackName(otherStackName)}-${otherStackNodeId}-${key}`;
    const marshalledResult = Cdk.Fn.importValue(globallyUniqueKeyExport);
    const result = Cdk.Fn.split(',', marshalledResult);
    return result;
  }

  /**
   * @brief fromVpcAttributes() that fetches the attributes using a naming convetion for stack
   * export/import.
   */
  static importFromStackOutput(scope: Cdk.Construct, id: string, otherStackName:string, otherStackNodeId: string): VpcPortable {

    // get the exported outputs from the other stack
    const vpcId = VpcPortable.getOtherStackOutput('vpcId', otherStackName, otherStackNodeId);
    const availabilityZones = VpcPortable.getOtherStackOutputList('availabilityZones', otherStackName, otherStackNodeId);
    const isolatedSubnets = VpcPortable.getOtherStackOutputList('isolatedSubnets', otherStackName, otherStackNodeId);
    const privateSubnets = VpcPortable.getOtherStackOutputList('privateSubnets', otherStackName, otherStackNodeId);
    const publicSubnets = VpcPortable.getOtherStackOutputList('publicSubnets', otherStackName, otherStackNodeId);
    const isolatedRouteTableIds = VpcPortable.getOtherStackOutputList('isolatedRouteTableIds', otherStackName, otherStackNodeId);
    const privateRouteTableIds = VpcPortable.getOtherStackOutputList('privateRouteTableIds', otherStackName, otherStackNodeId);
    const publicRouteTableIds = VpcPortable.getOtherStackOutputList('publicRouteTableIds', otherStackName, otherStackNodeId);

    const importedVpc = Ec2.Vpc.fromVpcAttributes(scope, id, {
      vpcId: vpcId,
      availabilityZones: availabilityZones,
      isolatedSubnetIds: isolatedSubnets,
      privateSubnetIds: privateSubnets,
      publicSubnetIds: publicSubnets,
      isolatedSubnetRouteTableIds: isolatedRouteTableIds,
      privateSubnetRouteTableIds: privateRouteTableIds,
      publicSubnetRouteTableIds: publicRouteTableIds,
    }) as VpcPortable;

    return importedVpc;
  }


  constructor(scope: Cdk.Construct, id: string, props?: Ec2.VpcProps) {
    super(scope, id, props);
  }

  /**
   * @brief publish a legal stack list output with globally unique key
   */
  publishStackOutputList(key: string, values: string[]) {

    const globallyUniqueKeyExport = `${VpcPortable.mangleStackName(this.stack.stackName)}-${this.node.id}-${key}`;
    const marshalledStringList = values.join(',');
    new Cdk.CfnOutput(this.stack, globallyUniqueKeyExport, {
      value: marshalledStringList,
      exportName: globallyUniqueKeyExport,
    });
  }

  /**
   * @brief publish a legal stack output with globally unique key
   */
  publishStackOutput(key: string, value: string) {
    const globallyUniqueKeyExport = `${VpcPortable.mangleStackName(this.stack.stackName)}-${this.node.id}-${key}`;
    new Cdk.CfnOutput(this.stack, globallyUniqueKeyExport, {
      value: value,
      exportName: globallyUniqueKeyExport,
    });
  }


  /**
   * @brief export Vpc and all components as stack output
   */
  exportToStackOutput() {

    // grab all the subnet IDs
    const isolatedSubnets = this.selectSubnets({
      subnetType: Ec2.SubnetType.ISOLATED,
    });
    const privateSubnets = this.selectSubnets({
      subnetType: Ec2.SubnetType.PRIVATE,
    });
    const publicSubnets = this.selectSubnets({
      subnetType: Ec2.SubnetType.PUBLIC,
    });

    // from subnets, grab all the route tables
    const isolatedRouteTableIds = this.isolatedSubnets.map((el: Ec2.ISubnet) => {
      return el.routeTable.routeTableId;
    });
    const privateRouteTableIds = this.privateSubnets.map((el: Ec2.ISubnet) => {
      return el.routeTable.routeTableId;
    });
    const publicRouteTableIds = this.publicSubnets.map((el: Ec2.ISubnet) => {
      return el.routeTable.routeTableId;
    });

    // Get the AZs
    const azs = this.availabilityZones;

    // publish
    this.publishStackOutput('vpcId', this.vpcId);
    this.publishStackOutputList('availabilityZones', azs);
    this.publishStackOutputList('isolatedSubnets', isolatedSubnets.subnetIds);
    this.publishStackOutputList('privateSubnets', privateSubnets.subnetIds);
    this.publishStackOutputList('publicSubnets', publicSubnets.subnetIds);
    this.publishStackOutputList('isolatedRouteTableIds', isolatedRouteTableIds);
    this.publishStackOutputList('privateRouteTableIds', privateRouteTableIds);
    this.publishStackOutputList('publicRouteTableIds', publicRouteTableIds);


  }

}
