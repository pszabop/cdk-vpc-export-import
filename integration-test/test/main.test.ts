import '@aws-cdk/assert/jest';
import * as util from 'util';
import { App } from '@aws-cdk/core';
import { SrcStack, DestStack } from '../src/main';

test('Snapshot', () => {
  const app = new App();
  const exportingStack = new SrcStack(app, 'testExport');
  const importingStack = new DestStack(app, 'testImport');

  console.log(util.inspect(exportingStack, false, 5));
  console.log(util.inspect(importingStack, false, 5));
  expect(exportingStack).not.toHaveResource('AWS::S3::Bucket');
  expect(importingStack).not.toHaveResource('AWS::S3::Bucket');
  expect(app.synth().getStackArtifact(exportingStack.artifactId).template).toMatchSnapshot();
  expect(app.synth().getStackArtifact(importingStack.artifactId).template).toMatchSnapshot();
});
