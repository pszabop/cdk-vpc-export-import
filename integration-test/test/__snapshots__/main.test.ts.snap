// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
Object {
  "Outputs": Object {
    "testExporttestVpcavailabilityZones": Object {
      "Export": Object {
        "Name": "testExport-testVpc-availabilityZones",
      },
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            Object {
              "Fn::Select": Array [
                0,
                Object {
                  "Fn::GetAZs": "",
                },
              ],
            },
            ",",
            Object {
              "Fn::Select": Array [
                1,
                Object {
                  "Fn::GetAZs": "",
                },
              ],
            },
          ],
        ],
      },
    },
    "testExporttestVpcisolatedRouteTableIds": Object {
      "Export": Object {
        "Name": "testExport-testVpc-isolatedRouteTableIds",
      },
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            Object {
              "Ref": "testVpcisolatedSubnet1RouteTable024A87DA",
            },
            ",",
            Object {
              "Ref": "testVpcisolatedSubnet2RouteTable91503A40",
            },
          ],
        ],
      },
    },
    "testExporttestVpcisolatedSubnets": Object {
      "Export": Object {
        "Name": "testExport-testVpc-isolatedSubnets",
      },
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            Object {
              "Ref": "testVpcisolatedSubnet1Subnet8954A677",
            },
            ",",
            Object {
              "Ref": "testVpcisolatedSubnet2Subnet5402C85D",
            },
          ],
        ],
      },
    },
    "testExporttestVpcprivateRouteTableIds": Object {
      "Export": Object {
        "Name": "testExport-testVpc-privateRouteTableIds",
      },
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            Object {
              "Ref": "testVpcprivateSubnet1RouteTable9E85C1BE",
            },
            ",",
            Object {
              "Ref": "testVpcprivateSubnet2RouteTableD02911C2",
            },
          ],
        ],
      },
    },
    "testExporttestVpcprivateSubnets": Object {
      "Export": Object {
        "Name": "testExport-testVpc-privateSubnets",
      },
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            Object {
              "Ref": "testVpcprivateSubnet1Subnet57A92184",
            },
            ",",
            Object {
              "Ref": "testVpcprivateSubnet2SubnetB7B5DCC7",
            },
          ],
        ],
      },
    },
    "testExporttestVpcpublicRouteTableIds": Object {
      "Export": Object {
        "Name": "testExport-testVpc-publicRouteTableIds",
      },
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            Object {
              "Ref": "testVpcpublicSubnet1RouteTableA92BEDE8",
            },
            ",",
            Object {
              "Ref": "testVpcpublicSubnet2RouteTable5620AB76",
            },
          ],
        ],
      },
    },
    "testExporttestVpcpublicSubnets": Object {
      "Export": Object {
        "Name": "testExport-testVpc-publicSubnets",
      },
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            Object {
              "Ref": "testVpcpublicSubnet1Subnet1DED695E",
            },
            ",",
            Object {
              "Ref": "testVpcpublicSubnet2Subnet8A5922AE",
            },
          ],
        ],
      },
    },
    "testExporttestVpcvpcId": Object {
      "Export": Object {
        "Name": "testExport-testVpc-vpcId",
      },
      "Value": Object {
        "Ref": "testVpcCB3A84F3",
      },
    },
  },
  "Resources": Object {
    "testVpcCB3A84F3": Object {
      "Properties": Object {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "testVpcIGW8765943D": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "testVpcVPCGWD6E219A3": Object {
      "Properties": Object {
        "InternetGatewayId": Object {
          "Ref": "testVpcIGW8765943D",
        },
        "VpcId": Object {
          "Ref": "testVpcCB3A84F3",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
    "testVpcisolatedSubnet1RouteTable024A87DA": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc/isolatedSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "testVpcCB3A84F3",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "testVpcisolatedSubnet1RouteTableAssociationF1704DC2": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "testVpcisolatedSubnet1RouteTable024A87DA",
        },
        "SubnetId": Object {
          "Ref": "testVpcisolatedSubnet1Subnet8954A677",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "testVpcisolatedSubnet1Subnet8954A677": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.0.0/22",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "isolated",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Isolated",
          },
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc/isolatedSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "testVpcCB3A84F3",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "testVpcisolatedSubnet2RouteTable91503A40": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc/isolatedSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "testVpcCB3A84F3",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "testVpcisolatedSubnet2RouteTableAssociation581D3C71": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "testVpcisolatedSubnet2RouteTable91503A40",
        },
        "SubnetId": Object {
          "Ref": "testVpcisolatedSubnet2Subnet5402C85D",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "testVpcisolatedSubnet2Subnet5402C85D": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.4.0/22",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "isolated",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Isolated",
          },
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc/isolatedSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "testVpcCB3A84F3",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "testVpcprivateSubnet1DefaultRoute5B5BBC62": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "testVpcpublicSubnet1NATGateway091FC1F0",
        },
        "RouteTableId": Object {
          "Ref": "testVpcprivateSubnet1RouteTable9E85C1BE",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "testVpcprivateSubnet1RouteTable9E85C1BE": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc/privateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "testVpcCB3A84F3",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "testVpcprivateSubnet1RouteTableAssociation824288FC": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "testVpcprivateSubnet1RouteTable9E85C1BE",
        },
        "SubnetId": Object {
          "Ref": "testVpcprivateSubnet1Subnet57A92184",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "testVpcprivateSubnet1Subnet57A92184": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.8.0/22",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc/privateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "testVpcCB3A84F3",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "testVpcprivateSubnet2DefaultRouteF84B2517": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "testVpcpublicSubnet2NATGatewayAE707CA3",
        },
        "RouteTableId": Object {
          "Ref": "testVpcprivateSubnet2RouteTableD02911C2",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "testVpcprivateSubnet2RouteTableAssociationFF59F07C": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "testVpcprivateSubnet2RouteTableD02911C2",
        },
        "SubnetId": Object {
          "Ref": "testVpcprivateSubnet2SubnetB7B5DCC7",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "testVpcprivateSubnet2RouteTableD02911C2": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc/privateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "testVpcCB3A84F3",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "testVpcprivateSubnet2SubnetB7B5DCC7": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.12.0/22",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc/privateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "testVpcCB3A84F3",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "testVpcpublicSubnet1DefaultRoute2568ED69": Object {
      "DependsOn": Array [
        "testVpcVPCGWD6E219A3",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "testVpcIGW8765943D",
        },
        "RouteTableId": Object {
          "Ref": "testVpcpublicSubnet1RouteTableA92BEDE8",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "testVpcpublicSubnet1EIPCFD44F97": Object {
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc/publicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "testVpcpublicSubnet1NATGateway091FC1F0": Object {
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "testVpcpublicSubnet1EIPCFD44F97",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "testVpcpublicSubnet1Subnet1DED695E",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc/publicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "testVpcpublicSubnet1RouteTableA92BEDE8": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc/publicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "testVpcCB3A84F3",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "testVpcpublicSubnet1RouteTableAssociation39286FFF": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "testVpcpublicSubnet1RouteTableA92BEDE8",
        },
        "SubnetId": Object {
          "Ref": "testVpcpublicSubnet1Subnet1DED695E",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "testVpcpublicSubnet1Subnet1DED695E": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.16.0/22",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc/publicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "testVpcCB3A84F3",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "testVpcpublicSubnet2DefaultRoute11C518BE": Object {
      "DependsOn": Array [
        "testVpcVPCGWD6E219A3",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "testVpcIGW8765943D",
        },
        "RouteTableId": Object {
          "Ref": "testVpcpublicSubnet2RouteTable5620AB76",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "testVpcpublicSubnet2EIPE46BA58A": Object {
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc/publicSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "testVpcpublicSubnet2NATGatewayAE707CA3": Object {
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "testVpcpublicSubnet2EIPE46BA58A",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "testVpcpublicSubnet2Subnet8A5922AE",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc/publicSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "testVpcpublicSubnet2RouteTable5620AB76": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc/publicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "testVpcCB3A84F3",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "testVpcpublicSubnet2RouteTableAssociationAE36E690": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "testVpcpublicSubnet2RouteTable5620AB76",
        },
        "SubnetId": Object {
          "Ref": "testVpcpublicSubnet2Subnet8A5922AE",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "testVpcpublicSubnet2Subnet8A5922AE": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.20.0/22",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "testExport/testVpc/publicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "testVpcCB3A84F3",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
  },
}
`;

exports[`Snapshot 2`] = `Object {}`;
